controllers/staff.js

```diff
@@ -50,6 +50,37 @@ LIMIT $4;
 
 const psGetMinMaxDate = new PS({ name: 'get-min-max-date', text: 'SELECT min(deliveredTime) as minDate, max(deliveredTime) as maxDate FROM Orders;' });
 
+const psGetPromoStats = new PS({ name: 'get-promo-stats', text: `
+WITH PromoDuration AS (
+SELECT pid, startDate, endDate, (endDate - startDate + 1) as duration
+FROM RestaurantPromos natural join Promotions
+WHERE rid = $1)
+
+SELECT pid, duration, count(*) as numOrders
+FROM PromoDuration natural join Orders
+WHERE orderTime >= startDate AND orderTime <= endDate
+GROUP BY pid, duration; 
+`});
+
+const psGetAllOrders = new PS({ name: 'get-all-orders', text: 
+`
+WITH OrderRestaurantPair AS (
+    SELECT DISTINCT rid, oid
+    FROM Collates natural join Orders
+),
+RidToName AS (
+    SELECT uid as riderId, name as rname
+    FROM Riders natural join Users
+),
+CidToName AS (
+    SELECT uid as customerId, name as cname
+    FROM Customers natural join Users
+)
+SELECT oid, rname, cname, orderTime, deliveredTime, finalPrice, (SELECT string_agg(fname, ', ') FROM COLLATES C WHERE C.oid = X.oid) as itemsOrdered
+FROM (Orders natural join OrderRestaurantPair join RidToName using (riderId) join CidToName using (customerId)) as X
```

Comment:
What's X?

Reply:
table alias. it is used in the line above to qualify oid